embedded/document/engine.go

```diff
@@ -852,3 +829,44 @@ func (e *Engine) BulkInsertDocuments(ctx context.Context, collectionName string,
 
 	return
 }
+
+func NewDocumentReader(reader sql.RowReader) DocumentReader {
+	return &documentReader{reader: reader}
+}
+
+type documentReader struct {
+	reader sql.RowReader
+}
+
+// ReadStructMessagesFromReader reads a number of messages from a reader and returns them as a slice of Struct messages.
+func (d *documentReader) Read(ctx context.Context, count int) ([]*structpb.Struct, error) {
```

Comment:
`ErrNoMoreDocuments` may be returned if less than specified documents can be read but still returning the documents that has been read. So it would behave the same way as when reading from an input stream with io.EOF

Alternatively the Read method be as in sql layer `Read(ctx context.Context` and return the next document

Reply:
added ErrNoMoreDocuments, keeping read method as it is for now to avoid loops while reading from the reader